From 8586a6814f54e6b48c0f11a3ffd7cc8a735d0272 Mon Sep 17 00:00:00 2001
From: dragonGR <alex@dragongr.dev>
Date: Fri, 5 Jun 2020 05:56:10 +0300
Subject: [PATCH 1/1] libstagefright: Fix memory leak

mDrmManagerClient is linked thanks to DrmInitialization() in case mDecryptHandle isn't reporting NULL.
This sounds like a good deal but mDrmManagerClient isn't cleared anywhere.

This was patched properly on Oreo but lost the way to Q.

Change-Id: Ib8c95e23b110748278001326360d73be007d4ff0
---
 media/libstagefright/http/MediaHTTP.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/media/libstagefright/http/MediaHTTP.cpp b/media/libstagefright/http/MediaHTTP.cpp
index 0fba3dc37..3e7f0a07e 100644
--- a/media/libstagefright/http/MediaHTTP.cpp
+++ b/media/libstagefright/http/MediaHTTP.cpp
@@ -69,6 +69,11 @@ void MediaHTTP::clearDRMState_l() {
         mDrmManagerClient->closeDecryptSession(mDecryptHandle);
         mDecryptHandle = NULL;
     }
+
+    if (mDrmManagerClient != NULL) {
+        delete mDrmManagerClient;
+        mDrmManagerClient = NULL;
+    }
 }
 
 }  // namespace android
-- 
2.27.0

